services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/authentication/TermsOfServiceManager.java

```diff
@@ -0,0 +1,138 @@
+package org.sagebionetworks.repo.manager.authentication;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.sagebionetworks.repo.model.auth.AuthenticationDAO;
+import org.sagebionetworks.repo.model.auth.TermsOfServiceInfo;
+import org.sagebionetworks.repo.model.auth.TermsOfServiceRequirements;
+import org.sagebionetworks.repo.model.utils.github.Release;
+import org.sagebionetworks.repo.util.github.GithubApiClient;
+import org.sagebionetworks.repo.util.github.GithubApiException;
+import org.sagebionetworks.schema.parser.ParseException;
+import org.sagebionetworks.schema.parser.SchemaIdParser;
+import org.sagebionetworks.util.Clock;
+import org.springframework.stereotype.Service;
+
+import com.google.common.base.Ticker;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.ExecutionError;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+
+@Service
+public class TermsOfServiceManager {
+	
+	private static final Logger LOGGER = LogManager.getLogger(TermsOfServiceManager.class);	
+	
+	private static final TermsOfServiceRequirements DEFAULT_REQUIRMENTS = new TermsOfServiceRequirements()
+		.setMinimumTermsOfServiceVersion("0.0.0")
+		.setRequirementDate(Date.from(Instant.parse("2011-01-01T00:00:00.000Z")));
+
+	private static final String VERSION_LATEST = "latest";
+	
+	private static final Duration VERSION_CACHE_EXPIRATION = Duration.of(10, ChronoUnit.MINUTES);
+	
+	static final String ORG = "Sage-Bionetworks";
+	static final String REPO = "Sage-Governance-Documents";
+	
+	private static final String TOS_URL_FORMAT = "https://raw.githubusercontent.com/" + ORG + "/" + REPO + "/refs/tags/%s/Terms.md";
+		
+	private AuthenticationDAO authDao;
+	
+	private GithubApiClient githubClient;
+	
+	private LoadingCache<String, Release> versionCache;
+	
+	private String latestVersionFallback;
+	
+	public TermsOfServiceManager(AuthenticationDAO authDao, GithubApiClient githubClient, Clock clock) {
+		this.authDao = authDao;
+		this.githubClient = githubClient;
+		this.versionCache = CacheBuilder.newBuilder()
+			.ticker(new Ticker() {
+				
+				@Override
+				public long read() {
+					return clock.nanoTime();
+				}
+			})
+			.expireAfterWrite(VERSION_CACHE_EXPIRATION)
+			.build(CacheLoader.from(this::fetchRelease));
+	}
+	
+	@PostConstruct
+	public void initialize() {
```

Comment:
maybe add the synchronize keyword or use an atomic refe or other thread safe tool.